indico/modules/events/surveys/controllers/management.py

```diff
@@ -149,3 +158,32 @@ def _process(self):
             flash(_('Question "{title}" updated').format(title=self.question.title), 'success')
             return jsonify_data(flash=False)
         return jsonify_template('events/surveys/edit_question.html', form=form, question=self.question)
+
+
+class RHDeleteSurveyQuestion(RHManageSurveysBase):
+    def _checkParams(self, params):
+        RHManageSurveysBase._checkParams(self, params)
+        self.question = SurveyQuestion.get_one(request.view_args['question_id'])
+        self.survey = Survey.get(request.view_args['survey_id'])
+
+    def _process(self):
+        db.session.delete(self.question)
+        db.session.flush()
+        flash(_('Question {} successfully deleted'.format(self.question.title)), 'success')
+        return redirect(url_for('.manage_questionnaire', self.survey))
+
+
+class RHChangeQuestionPosition(RHManageSurveysBase):
+    def _checkParams(self, params):
+        RHManageSurveysBase._checkParams(self, params)
+        self.survey = Survey.get(request.view_args['survey_id'])
+
+    def _process(self):
+        questions = {question.id: question for question in self.survey.questions}
+        question_ids_in_order = json.loads(self.getRequestParams()['questionIdsInOrder'])
```

Comment:
Never use those legact methods. `json.loads(request.form['...'])` is much better.